src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2280,4 +2275,17 @@ private static String maybeQuoteETag(String eTag) {
     private static boolean startsWithIgnoreCase(String string, String prefix) {
         return string.toLowerCase().startsWith(prefix.toLowerCase());
     }
+
+    // Protects against illegal characters in blob names only if the client requests it
+    private String encodeBlob(String encodingType, String blobName) {
+        if (encodingType != null && encodingType.equals("url")) {
+            try {
+                return URLEncoder.encode(blobName, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                return blobName;
```

Comment:
Can you `throw Throwables.propagate(e)` since this exception should never be thrown?